### DISCUSSION TITLE
IP Range for shields

### DISCUSSION THREAD

**Discussion by gbcham1** [badges/shields#8812] [Q&A] [answered]
Hi, I'm trying to integrate shields into our GH repos but all of the services that we run which we want metrics for are hosted internally. I want to open up a hole in the firewall to send to an internal API which can fetch info from our services and return them as JSON to shields, but to do this I need to know which IP addresses to whitelist. GitHub, Jira, and many other sites have their IP ranges made public for this purpose but I can't find the IP range for shields anywhere. 

I've tried setting up an http server and sending requests to it via shields and so far I've had 4 different IP addresses, 3 of them wildly different and I don't want to keep doing this over and over again until I have a rough idea of the IP range:
66.225.222.66
66.225.222.67
145.40.64.161
159.203.192.15

Anyone know if there is a strictly defined IP range available anywhere? or a meta url which can provide them?

**Comment by chris48s** [comment, marked as answer]
We don't publish IP ranges or make any guarantees about where traffic is going to come from. We are hosted on fly.io. The IPs traffic comes from are subject to change by the upstream provider and we don't really have any control over it.

We could also in future change which regions we're deployed in or migrate to a different hosting provider.

If you need to use shields badges with services that are behind a corporate firewall, the most common solution to this is to self-host your own instance of shields inside the firewall and use that rather than shields.io. There are docs on how to do this at https://github.com/badges/shields/blob/master/doc/self-hosting.md
